core/Test/Tasty/Parallel.hs

```diff
@@ -52,9 +53,9 @@ runInParallel nthreads actions = do
 
   actionsVar <- atomically $ newTMVar actions
 
-  pids <- replicateM nthreads (forkIO $ work actionsVar)
+  pids <- replicateM nthreads (async $ work actionsVar)
 
-  return $ mapM_ killThread pids
+  return $ mapM_ cancel pids
```

Comment:
I believe the reason the `bracket` block doesn't release its resources is because the main thread terminates too early. So my suggested fix is to switch to `async` (which is already a dependency) so we can benefit from `cancel`'s guarantee that it only returns once the thread has completed, that is, once all the resources are released.